inferno/net.py

```diff
@@ -502,17 +504,26 @@ def fit_loop(self, X, y, epochs=None):
         self.check_data(X, y)
         epochs = epochs or self.max_epochs
         optimizer = self.get_optimizer()
+        if self.train_split:
+            X_train, X_valid, y_train, y_valid = self.train_split(X, y)
+        else:
+            X_train, X_valid, y_train, y_valid = X, None, y, None
+
         for epoch in range(epochs):
             self.notify('on_epoch_begin', X=X, y=y)
 
-            for xi, yi in self.get_iterator(X, y, train=True):
+            for xi, yi in self.get_iterator(X_train, y_train, train=True):
                 self.notify('on_batch_begin', X=xi, y=yi, train=True)
                 loss = self.train_step(xi, yi, optimizer)
                 self.history.record_batch('train_loss', loss.data[0])
                 self.history.record_batch('train_batch_size', len(xi))
                 self.notify('on_batch_end', X=xi, y=yi, train=True)
 
-            for xi, yi in self.get_iterator(X, y, train=False):
+            if X_valid is None:
+                self.notify('on_epoch_end', X=X, y=y)
+                continue
```

Comment:
I'm not a fan of repeating the callback call here. I would argue for wrapping the validation loop instead.